pybaseball/datahelpers/postprocessing.py

```diff
@@ -18,24 +18,58 @@
     (re.compile(r'^\d{4}-\d{1,2}-\d{1,2}T\d{2}:\d{2}:\d{2}.\d{1,6}Z$'), '%Y-%m-%dT%H:%M:%S.%fZ'),
 ]
 
+
 def try_parse_dataframe(
     data: pd.DataFrame,
     null_replacement: Union[str, int, float, datetime] = np.nan,
     known_percentages: List[str] = []
 ) -> pd.DataFrame:
-    values = [
-        {column: try_parse(data[column][row_i], column) for column in data.columns}
-        for row_i in range(len(data)) 
-    ]
+    data_copy = data.copy()
+    data_copy = coalesce_nulls(data_copy, null_replacement)
+    data_copy = data_copy.apply(pd.to_numeric, errors='ignore', downcast='integer').convert_dtypes(convert_string=False)
+
+    rows = len(data_copy.values)
+
+    for column in data_copy.columns:
+        if str(data_copy[column].dtype) in ['object', 'string']:
+            # Only check the first value of the column and test that;
+            # this is faster than blindly trying to convert entire columns
+            first_value_index = data_copy[column].first_valid_index()
+            if first_value_index is None:
+                # All nulls
+                continue
+            first_value = data_copy[column].loc[first_value_index]
+
+            if str(first_value).endswith('%') or column.endswith('%') or column in known_percentages:
+                column_pos = data_copy.columns.get_loc(column)
+                for row in range(rows):
+                    data_copy.iat[row, column_pos] = try_parse(
```

Comment:
I don't think you need to use the `try_parse` function anymore, since you know you're trying to parse percentages. It's also redundant to pass `column` to `try_parse` when you've already checked the column. Maybe we need a new function here, `parse_percentage` that's more limited in scope? 

even better would be to use pandas vectorized operations to avoid the explicit loop over rows. maybe something like, 
```
data_copy[column] = data_copy[column].str.replace("%", " ").astype(float) / 100
```
?